test/OpenIddict.Abstractions.Tests/Primitives/OpenIddictExtensionsTests.cs

```diff
@@ -0,0 +1,872 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/openiddict/openiddict-core for more information concerning
+ * the license and the contributors participating to this project.
+ */
+
+using System;
+using Xunit;
+
+namespace OpenIddict.Abstractions.Tests.Primitives
+{
+    public class OpenIdConnectExtensionsTests
+    {
+        [Fact]
+        public void GetAcrValues_ThrowsAnExceptionForNullRequest()
+        {
+            // Arrange
+            var request = (OpenIddictRequest) null;
+
+            // Act and assert
+            var exception = Assert.Throws<ArgumentNullException>(() => request.GetAcrValues());
+
+            Assert.Equal("request", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData(null, new string[0])]
+        [InlineData("mod-pr", new[] { "mod-pr" })]
+        [InlineData("mod-pr ", new[] { "mod-pr" })]
+        [InlineData(" mod-pr ", new[] { "mod-pr" })]
+        [InlineData("mod-pr mod-mf", new[] { "mod-pr", "mod-mf" })]
+        [InlineData("mod-pr     mod-mf", new[] { "mod-pr", "mod-mf" })]
+        [InlineData("mod-pr mod-mf ", new[] { "mod-pr", "mod-mf" })]
+        [InlineData(" mod-pr mod-mf", new[] { "mod-pr", "mod-mf" })]
+        [InlineData("mod-pr mod-pr mod-mf", new[] { "mod-pr", "mod-mf" })]
+        [InlineData("mod-pr MOD-PR mod-mf", new[] { "mod-pr", "MOD-PR", "mod-mf" })]
+        public void GetAcrValues_ReturnsExpectedAcrValues(string value, string[] values)
+        {
+            // Arrange
+            var request = new OpenIddictRequest
+            {
+                AcrValues = value
+            };
+
+            // Act
+            var actualValues = request.GetAcrValues();
```

Comment:
Are these changes required because we now return immutable collections, where the order is not respected?

Reply:
I opened https://github.com/openiddict/openiddict-core/issues/846 to track future improvements to revert to ASOS' behavior, where the returned collections respected the order set by the client app.

Reply:
Yes, ImmutableHashSet<T> is unordered set, what  failed unit tests and that's why I changed the tests

Reply:
Great, I'll update the tests once we move back to collection types that respect the user-defined order.